lib/parachain/service.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachain
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+)
+
+type Service struct {
+	Network Network
+}
+
+func NewService(net Network, genesisHash common.Hash) (*Service, error) {
+	// TODO: Where do I get forkID and version from from?
```

Comment:
Maybe include an issue number here if related to a different issue?

Reply:
created constants for version, as there is just one version. And created an issue for fetching fork id. Looks like we will have to fetch it from chain spec, https://github.com/ChainSafe/gossamer/issues/3373